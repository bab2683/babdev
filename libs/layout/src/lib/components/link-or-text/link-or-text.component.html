<a
  class="link-effect-1"
  target="_blank"
  [href]="source[key].link"
  *ngIf="source && source[key] && source[key].link; else simpleText"
>
  <ng-container *ngTemplateOutlet="content"></ng-container>
  {{ source[key].name }}
</a>
<ng-template #simpleText>
  <ng-container *ngTemplateOutlet="content"></ng-container>
  {{ source[key].name }}
</ng-template>

<ng-template #content>
  <ng-content></ng-content>
</ng-template>
