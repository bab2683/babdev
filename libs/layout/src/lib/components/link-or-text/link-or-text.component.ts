import { Component, Input } from '@angular/core';

@Component({
  selector: 'babdev-link-or-text',
  templateUrl: './link-or-text.component.html',
  styleUrls: ['./link-or-text.component.scss']
})
export class LinkOrTextComponent {
  @Input() key: string;
  @Input() source: { [key: string]: { link: string; name: string } };
}
