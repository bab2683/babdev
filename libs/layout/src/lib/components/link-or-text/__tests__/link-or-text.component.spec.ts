import { ComponentFixture, TestBed } from '@angular/core/testing';

import { LinkOrTextComponent } from '../link-or-text.component';

describe('LinkOrTextComponent', () => {
  let component: LinkOrTextComponent;
  let fixture: ComponentFixture<LinkOrTextComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [LinkOrTextComponent]
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(LinkOrTextComponent);
    component = fixture.componentInstance;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    component.source = {
      test: {
        link: 'https://angular.io/',
        name: 'angular'
      }
    };
    component.key = 'test';
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should match snapshot', () => {
    component.source = {
      test: {
        link: '',
        name: 'empty'
      }
    };
    component.key = 'test';

    expect(fixture).toMatchSnapshot();
  });

  it('should match snapshot', () => {
    component.source = {
      test: {
        link: '',
        name: 'empty'
      }
    };
    component.key = 'test2';

    expect(fixture).toMatchSnapshot();
  });
});
