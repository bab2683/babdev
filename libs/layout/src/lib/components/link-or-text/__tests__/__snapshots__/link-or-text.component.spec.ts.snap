// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkOrTextComponent should match snapshot 1`] = `
<babdev-link-or-text
  key={[Function String]}
  source={[Function Object]}
>
  <a
    class="link-effect-1"
    href="https://angular.io/"
    target="_blank"
  >
    
    
     angular
  
  </a>
</babdev-link-or-text>
`;

exports[`LinkOrTextComponent should match snapshot 2`] = `
<babdev-link-or-text
  key={[Function String]}
  source={[Function Object]}
>
  
</babdev-link-or-text>
`;

exports[`LinkOrTextComponent should match snapshot 3`] = `
<babdev-link-or-text
  key={[Function String]}
  source={[Function Object]}
>
  
</babdev-link-or-text>
`;
